File: WEB-INF/classes/bobby/eg1.java
package bobby;
import com.springboot.clone.annotations.*;
import com.springboot.clone.pojo.*;
@Path(value = "/student")
@Get
@InjectSessionScope
@InjectApplicationScope
@InjectApplicationDirectory
public class eg1
{
	private SessionScope sessionScope;
	private ApplicationScope applicationScope;
	private ApplicationDirectory applicationDirectory;
	@Autowired(name = "MANAN")
	private String manan;
	@Autowired(name = "TEST1")
	private String test1;
	
	@Forward(value = "/teacher")
	@Path(value = "/getStudent")
	public void getStudent()
	{
		System.out.println("This is a get student method");
		System.out.println("This Session Scope Name : "+sessionScope.getAttribute("Name"));
		System.out.println("Path : "+applicationDirectory.getDirectory().getPath());
		System.out.println("AUTOWIRED TESTING : "+test1+"     "+manan);
	}
	@Path(value = "/delete")
	public void deleteStudent()
	{
		System.out.println("Setting Naam Shabana");
		applicationScope.setAttribute("Naam", "Shabana");
	}
	@Path(value = "/edit")
	public Object editStudent()
	{
		return applicationScope.getAttribute("Naam");
	}
	@Forward(value = "/teacher/add")
	@Path(value = "/addStudent")
	public int addStudent()
	{
		return 1;
	}
	@OnStartup(priority = 2)
	public int get()
	{
		return 2;
	}
}

File: WEB-INF/js/eg1.js
class eg1Service
{
	editStudent()
	{
		var promise = new Promise((resolve,reject)=>
		{
			var xmlHttpRequest = new XMLHttpRequest();
			xmlHttpRequest.onreadystatechange = ()=>
			{
				if(xmlHttpRequest.readyState == 4)
				{
					if(xmlHttpRequest.status == 200)
					{
						var response = xmlHttpRequest.responseText;
						resolve(response);
					}
					else
					{
						reject();
					}
				}
			}
			xmlHttpRequest.open('GET', 'bobby/student/edit',true);
			xmlHttpRequest.send();
		});
		return promise;
	}
	addStudent()
	{
		var promise = new Promise((resolve,reject)=>
		{
			var xmlHttpRequest = new XMLHttpRequest();
			xmlHttpRequest.onreadystatechange = ()=>
			{
				if(xmlHttpRequest.readyState == 4)
				{
					if(xmlHttpRequest.status == 200)
					{
						var response = xmlHttpRequest.responseText;
						resolve(response);
					}
					else
					{
						reject();
					}
				}
			}
			xmlHttpRequest.open('GET', 'bobby/student/addStudent',true);
			xmlHttpRequest.send();
		});
		return promise;
	}
	deleteStudent()
	{
		var promise = new Promise((resolve,reject)=>
		{
			var xmlHttpRequest = new XMLHttpRequest();
			xmlHttpRequest.onreadystatechange = ()=>
			{
				if(xmlHttpRequest.readyState == 4)
				{
					if(xmlHttpRequest.status == 200)
					{
						var response = xmlHttpRequest.responseText;
						resolve(response);
					}
					else
					{
						reject();
					}
				}
			}
			xmlHttpRequest.open('GET', 'bobby/student/delete',true);
			xmlHttpRequest.send();
		});
		return promise;
	}
	getStudent()
	{
		var promise = new Promise((resolve,reject)=>
		{
			var xmlHttpRequest = new XMLHttpRequest();
			xmlHttpRequest.onreadystatechange = ()=>
			{
				if(xmlHttpRequest.readyState == 4)
				{
					if(xmlHttpRequest.status == 200)
					{
						var response = xmlHttpRequest.responseText;
						resolve(response);
					}
					else
					{
						reject();
					}
				}
			}
			xmlHttpRequest.open('GET', 'bobby/student/getStudent',true);
			xmlHttpRequest.send();
		});
		return promise;
	}
}
